src/main/java/io/vertx/spi/cluster/impl/zookeeper/ZKAsyncMultiMap.java

```diff
@@ -1,31 +1,37 @@
 package io.vertx.spi.cluster.impl.zookeeper;
 
 import io.vertx.core.*;
+import io.vertx.core.net.impl.ServerID;
 import io.vertx.core.spi.cluster.AsyncMultiMap;
 import io.vertx.core.spi.cluster.ChoosableIterable;
 import org.apache.curator.framework.CuratorFramework;
 import org.apache.curator.framework.recipes.cache.ChildData;
 import org.apache.curator.framework.recipes.cache.TreeCache;
+import org.apache.curator.framework.recipes.cache.TreeCacheEvent;
+import org.apache.curator.framework.recipes.cache.TreeCacheListener;
 
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Map;
-import java.util.Optional;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
 
 /**
  * Created by Stream.Liu
  */
 class ZKAsyncMultiMap<K, V> extends ZKMap<K, V> implements AsyncMultiMap<K, V> {
-  private TreeCache curatorCache;
+
+  private TreeCache treeCache;
+  private ConcurrentMap<String, ChoosableSet<V>> cache = new ConcurrentHashMap<>();
+  private final static String VERTX_SUBS = "__vertx.subs";
 
   ZKAsyncMultiMap(Vertx vertx, CuratorFramework curator, String mapName) {
     super(curator, vertx, ZK_PATH_ASYNC_MULTI_MAP, mapName);
-
-    // /io.vertx/asyncMultiMap/subs
-    curatorCache = new TreeCache(curator, mapPath);
+    treeCache = new TreeCache(curator, mapPath);
+    //we only make a listener for the path of __vertx.subs
+    if (mapName.equals(VERTX_SUBS)) treeCache.getListenable().addListener(new Listener());
```

Comment:
The listener addition shouldn't be conditional on the specific map name, otherwise, the TreeCache is superfluous.